src/OpenTelemetry.AutoInstrumentation/Loading/LazyInstrumentationLoader.cs

```diff
@@ -63,8 +63,8 @@ private class OnAssemblyLoadInitializer
         private static readonly ILogger Logger = OtelLogging.GetLogger();
         private readonly InstrumentationInitializer _instrumentationInitializer;
         private readonly LazyInstrumentationLoader _manager;
+        private readonly object _requiredAssembliesLocker = new();
```

Comment:
It would be fine to lock on `_requiredAssemblies`.
We would gain a little memory + it would be more clear that `_requiredAssemblies` is the shared resource.

Reference: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/statements/lock#guidelines

> lock on [...] instance that is unlikely to be used as a lock object by unrelated parts of the code

Reply:
Fixed.